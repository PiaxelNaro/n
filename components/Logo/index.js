// @flow
import styled, { css } from 'styled-components';

const Logo = styled.div`
    width: ${props => (props.mini ? '50px' : '100px')};
    height: ${props => (props.mini ? '50px' : '100px')};
    background-image: url(/static/logo.png);
    background-repeat: no-repeat;
    background-position: center;
    background-size: ${props => (props.mini ? '30px' : '60px')};
`;

export default Logo;
