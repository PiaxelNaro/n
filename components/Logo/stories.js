import React from 'react';
import { storiesOf } from '@kadira/storybook';
import styled from 'styled-components';
import Logo from '.';


const Wrapper = styled.div`
    background: #102235;
    width: 200px;
`;

storiesOf('Logo', module)
    .add('default', () => (
        <Wrapper>
            <Logo />
        </Wrapper>
    ))
    .add('mini', () => (
        <Wrapper>
            <Logo mini />
        </Wrapper>
    ))
    .add('mini & premium', () => (
        <Wrapper>
            <Logo mini premium />
        </Wrapper>
    ));
