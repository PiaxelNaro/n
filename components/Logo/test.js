import React from 'react';
import { shallow } from 'enzyme';
import Logo from '.';

describe('<Logo>', () => {
    it('default', () => {
        expect(
            shallow(<Logo />)
        ).toMatchSnapshot();
    });

    it('mini', () => {
        expect(
            shallow(<Logo mini />)
        ).toMatchSnapshot();
    });

    it('mini & premium', () => {
        expect(
            shallow(<Logo mini premium />)
        ).toMatchSnapshot();
    });
});
