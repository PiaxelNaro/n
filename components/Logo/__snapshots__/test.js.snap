// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> default 1`] = `
<div
  className="Logo__Logo-q7fvn9-0 efUBwB"
/>
`;

exports[`<Logo> mini & premium 1`] = `
<div
  className="Logo__Logo-q7fvn9-0 hWTYHT"
/>
`;

exports[`<Logo> mini 1`] = `
<div
  className="Logo__Logo-q7fvn9-0 hWnkCR"
/>
`;
